src/java/org/apache/cassandra/cql3/statements/ModificationStatement.java

```diff
@@ -815,16 +817,37 @@ final void addUpdates(UpdatesCollector collector,
                 }
                 else
                 {
+                    // Clustering keys need to be checked on their own
                     for (Clustering<?> clustering : clusterings)
                     {
                         clustering.validate();
+                        checkClusteringConstraints(clustering);
```

Comment:
See? Here you haven't put `clustering.validate` into `checkClusteringConstraints` as above.